### PR TITLE
Add Match Page

### PR DESCRIPTION
**Pull request by McNaBry**
## Description

Add a basic page for choosing matching criteria and dialogs for finding and retrying to find a match.

## Checklist

- [x] I have updated documentation
- [x] All tests passing

### CODE REVIEW

**Comment by limcaaarl** [comment]
Hi just a quick note, our wireframe for match page is slightly outdated. We're no longer including programming language as part of the matching criteria.

**Comment by McNaBry** [comment]
> Hi just a quick note, our wireframe for match page is slightly outdated. We're no longer including programming language as part of the matching criteria.
> 
> 
> 
> Edit:
> 
> The UI looks really good 🔥🔥🔥

Haha oops I'll remove it. Thanks for the review 😄

File: frontend/src/app/matching/matching.component.html [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    placeholder="No Difficulty Selected"
+                    [showClear]="true"
+                    class="p-fluid" />
+            </div>
+
+            <p-button
+                type="submit"
+                [label]="isProcessingMatch ? '' : 'Start Matching'"
+                [loading]="isProcessingMatch"
+                [disabled]="userCriteria.topics === null && userCriteria.difficulty === null"
```
Comment (samuelim01) [inline frontend/src/app/matching/matching.component.html:48]: Perhaps you could initialize `userCriteria.topics` as `[]` and check if `userCriteria.topics.length === 0`?

**Review by samuelim01 — APPROVED** [review]
LGTM! Sick work 🔥

File: frontend/src/app/matching/matching.component.html (hunk 1 above)
Comment (McNaBry) [reply frontend/src/app/matching/matching.component.html:48]: Weird thing is that when I clear the input or deselect all options via the PrimeNG component, the value of topics gets set to null. So I just opted to set null as the default state

Comment (McNaBry) [reply frontend/src/app/matching/matching.component.html:48]: I think I'll just use the `onClear` event to set topics to `[]` instead.

**Pull request merged**